# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/cookiestore/CookieStoreNotifier.h
# Commit: 2081d42e6ccd
# Full Hash: 2081d42e6ccdcc39b3d1b2e6523a8d44fed85390
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2024-09-13 04:28:43
# Regressor Bug: 1475599
# File Overlap Count: 1
# Description:
#   Bug 1475599 - part 8 - CookieStore API - e10s support, r=smaug,cookie-reviewers,timhuang
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D217574
# ==============================================================================

diff -r c6e20b939a21 -r 2081d42e6ccd dom/cookiestore/CookieStoreNotifier.h
--- a/dom/cookiestore/CookieStoreNotifier.h	Thu Sep 12 16:48:21 2024 +0000
+++ b/dom/cookiestore/CookieStoreNotifier.h	Thu Sep 12 16:48:21 2024 +0000
@@ -9,6 +9,7 @@
 
 #include "nsIObserver.h"
 #include "mozilla/OriginAttributes.h"
+#include "mozilla/MoveOnlyFunction.h"
 
 class nsISerialEventTarget;
 
@@ -38,6 +39,8 @@
   void AddObserversOnMainThread(bool aPrivateBrowsing);
   void RemoveObserversOnMainThread(bool aPrivateBrowsing);
 
+  void DispatchEvent(const CookieListItem& aItem, bool aDeletedEvent);
+
   // Raw pointer because this object is kept alive by this CookieStore object.
   CookieStore* mCookieStore;
 